tcms/core/tests/test_ajax.py

```diff
@@ -33,6 +42,45 @@ def test_empty_string_is_converted_to_bool(self):
         response = self.client.get(url)
         self.assertEqual(200, response.status_code)
 
+    def test_with_ulli_format(self):
+        """ When a request comes with info_type=categories, and format=ulli for given product_id,
+            we expect to receive all categories for that product, rendered as HTML unordered list
+        """
+
+        url = '%s?info_type=categories&product_id=%s&format=ulli' % (reverse('ajax-info'),
+                                                                     self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertContains(response, '<li>' + self.category_one.name + '</li>')
+        self.assertContains(response, '<li>' + self.category_two.name + '</li>')
+
+    def test_with_json_format(self):
+        """ When a request comes with info_type=categories for given product_id,
+            we expect to receive all categories for that product as array of JSON objects """
+
+        url = "%s?info_type=categories&product_id=%s" % (reverse('ajax-info'), self.product.pk)
+
+        response = self.client.get(url)
+
+        self.assertContains(response,
+                            '{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                            % (self.category_one.pk, self.category_one.name))
+        self.assertContains(response,
+                            '{"model": "testcases.category", "pk": %s, "fields": {"name": "%s"}}'
+                            % (self.category_two.pk, self.category_two.name))
+
+    def test_with_unrecognisable_infotype(self):
+        """ When a request comes with invalid info_type,
+            we expect to receive response containing the 'Unrecongnizable infotype' error message
+        """
+
+        url = "%s?info_type=INVALID" % reverse('ajax-info')
+
+        response = self.client.get(url)
+
+        self.assertEqual(response.content, b'Unrecognizable infotype')
```

Comment:
`str(response.content, encoding=settings.DEFAULT_CHARSET)` instead of comparing to a byte string.